### ISSUE TITLE
Multiline arguments violation with multiple trailing closures

### ISSUE THREAD

**Issue by jakubkiermasz-zd** [realm/SwiftLint#3605]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

For below code
```swift
talkProvider.getAvailability(forUser: userId) { [weak self] response in
    self?.handleSuccess(response)
} onError: { [weak self] _ in
    self?.handleError()
}
```

I receive this warning:
`Multiline Arguments Violation: Arguments should be either on the same line, or one per line. (multiline_arguments)`

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Pods
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? nope
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.4 Build version 12D4e
* Do you have a sample that shows the issue? Yes, it is. `Done linting! Found 1 violation, 0 serious in 1 file.`

**Comment by Sonal-Kachare** [comment]
To resolve this warning I did following changes to function declaration

Existing function declaration: 
```swift
func getAvailability(forUser: Int, _ onSuccess: @escaping (String) -> Void, onError: (String) -> Void) { }
```

Updated function declaration
```swift
func getAvailability(forUser: Int, onSuccess: @escaping (String) -> Void, onError: (String) -> Void) { }
```

Usage of updated function:
```swift
getAvailability(forUser: 10, onSuccess: { (_) in

}, onError: { (_) in

})
```

**Comment by jakubkiermasz-zd** [comment]
@Sonal-Kachare Thank you for your answer!
Of course, the code you've proposed is correct, but unfortunately, it doesn't use the trailing closure feature, which is powerful and recommended in Swift 5.3+.

**Comment by holgerde** [comment]
When you use a SwiftUI TextField, Xcode will generate this code for you:

```swift
TextField("Title", text: $text) { isEditing in
    // ...
} onCommit: {
    // ...
}
```

So this is definitely the recommended syntax and should not trigger a multiline arguments violation.

**Comment by jakubkiermasz-zd** [comment]
@marcelofabri could you take a look at this, please :)?

**Comment by dnadoba** [comment]
I run into a similar issue with multiple trailing closures but for the `vertical_parameter_alignment_on_call` rule:
```swift
PreviewView(
    previewLayout: .fitWidth(360)
) {
    SomeView()
} update: {
    $0.update(data: data)
}
```
Which creates the following warning: 
> Vertical Parameter Alignment On Call Violation: Function parameters should be aligned vertically if they're in multiple lines in a method call. (vertical_parameter_alignment_on_call)

**Comment by macintosh-HD** [comment]
Any news on the status of this issue?
I'm still getting this warning in version 0.47.1.

**Comment by edelabar** [comment]
Also seeing this in 0.47.1.

Custom signature:

``` swift
init(
	_ title: String,
	@ViewBuilder leftButtonBuilder: () -> LeftButton,
	@ViewBuilder rightButtonBuilder: () -> RightButton
)
```

Failing Invocation (`multiline_arguments`):

``` swift
TitleBar("Profile") {
	backButton
} rightButtonBuilder: {
	editButton
}
```

Also tried...

Fails `multiline_arguments_brackets` and `vertical_parameter_alignment_on_call`:
``` swift 
TitleBar("Profile")
{
	backButton
} rightButtonBuilder: {
	editButton
}
```

Works, but not ideal:

``` swift
TitleBar(
	"Profile",
	leftButtonBuilder: {
		backButton
	}, rightButtonBuilder: {
		editButton
	}
)
```

**Comment by yo1995** [comment]
It seems like a workaround would be config it differently:

```yml
multiline_arguments:
  only_enforce_after_first_closure_on_first_line: true
```

Please refer to https://realm.github.io/SwiftLint/multiline_arguments.html

**Comment by marcelofabri** [comment]
This should've been fixed by https://github.com/realm/SwiftLint/pull/4750

**Comment by joohae-kim** [comment]
@marcelofabri the issue is still happening on 0.55.1
